src/CompositeChat.cpp

```diff
@@ -45,10 +45,10 @@ cCompositeChat::~cCompositeChat()
 
 void cCompositeChat::Clear(void)
 {
-	for (cParts::iterator itr = m_Parts.begin(), end = m_Parts.end(); itr != end; ++itr)
+	for (auto & Part : m_Parts)
 	{
-		delete *itr;
-	}  // for itr - m_Parts[]
+		Part.reset();
+	}
 	m_Parts.clear();
```

Comment:
Don't need to explicitly reset the unique_ptrs.

Reply:
hm, so we don't need a explicit destructor?

Reply:
`clear()` destroys all the elements automatically.